lib/copy_color_spinor_offset.cu

```diff
@@ -22,27 +22,63 @@ namespace quda
     }
   }
 
-  template <class Float, int nColor, int nSpin>
-  void copy_color_spinor_offset(ColorSpinorField &out, const ColorSpinorField &in, CommKey offset, QudaPCType pc_type)
+  template <class Float, int nColor>
```

Comment:
I'm not sure I follow the changes in this file.  Why was it necessary to make separate spin-dependent functions here, as opposed to the previous single templated function?

Reply:
So this may be just me not knowing the code. But originally it supported `QUDA_SPACE_COLOR_SPIN_FIELD_ORDER` and `QUDA_SPACE_SPIN_COLOR_FIELD_ORDER` and I didn't know if it supported specifically the internal orderings of 'QUDA_FLOAT_FIELD_ORDER` or `QUDA_FLOAT_2_FIELD_ORDER` etc. So I wanted to add those. But Staggered only supports spin 1 which is a subset of these. So I decided to make an Ns4 and an Ns1 specialization. 

But then I did more faffing so it may be that I can revert this and it will still work... I will check

Reply:
The original file worked. So I reverted to that. I guess this was when I was messing around with mappers and I did something odd.